app/Example.java

```diff
@@ -0,0 +1,50 @@
+import android.view.View;
+
+import io.ably.lib.realtime.AblyRealtime;
+import io.ably.lib.realtime.Channel;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.Message;
+import io.ably.lib.types.Param;
+
+public class Example {
+
+	private static final String API_KEY = "INSERT-YOUR-API-KEY-HERE";
+
+	public void publishAndSubcscribe(Button buttonPublish, Button buttonHistory) {
+		AblyRealtime realtime = new AblyRealtime(API_KEY);
+        /* Get a channel in the "persisted" namespace */
+        final Channel channel = realtime.channels.get("persisted:sounds");
+
+        /* Publish three messages when the publish button is clicked */
+        buttonPublish.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                /* Surround Channel#publish with try/catch */
+                try {
+                    channel.publish("play", "bark");
+                    channel.publish("play", "meow");
+                    channel.publish("play", "cluck");
+                } catch(AblyException exception){
+                    exception.printStackTrace();
+                }
+            }
+        });
+
+        /* Retrieve messages from history after click*/
+        buttonHistory.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    Param[] options = new Param[]{ new Param("limit", "3") };
+                    /* Insert parameters to retrieve last three messages, can insert null if no parameters are needed */
+                    Message[] messages = channel.history(options).items();
```

Comment:
Why no feedback of what is happening i.e. similar to https://github.com/ably/tutorials/blob/history-ruby/example.rb#L15 so that the user sees a request starting and then the respone?